spec/models/timecard_spec.rb

```diff
@@ -19,60 +55,182 @@
+             but with an end_time; a generic timecard' do
+      before(:example) do
+        @working_timecard  = create(:timecard, person: @cj, start_time: Time.current,
+                                    end_time: nil, status: 'Incomplete')
+        @finished_timecard = create(:timecard, person: @cj, start_time: 1.hour.ago,
+                                    end_time: Time.current, status: 'Incomplete')
+        @timecard          = create(:timecard, person: @cj)
+      end
+
+      it 'returns the timecard with Incomplete status and no end_time' do
+        expect(described_class.working).to contain_exactly(@working_timecard)
+      end
+    end
+  end
+
+  describe 'active scope' do
+    it 'returns a chainable relation' do
+      expect(described_class.active).to be_a_kind_of(ActiveRecord::Relation)
+    end
+    
     it 'returns all timecards that are not cancelled' do
       incomplete_timecard = create(:timecard, person: @cj, status: 'Incomplete')
       unverified_timecard = create(:timecard, person: @cj, status: 'Unverified')
       error_timecard = create(:timecard, person: @cj, status: 'Error')
       verified_timecard = create(:timecard, person: @cj, status: 'Verified')
       cancelled_timecard = create(:timecard, person: @cj, status: 'Cancelled')
       active_timecards = Timecard.active
-      expect(active_timecards).to include(incomplete_timecard)
-      expect(active_timecards).to include(unverified_timecard)
-      expect(active_timecards).to include(error_timecard)
-      expect(active_timecards).to include(verified_timecard)
+      expect(active_timecards).to include(incomplete_timecard, unverified_timecard, error_timecard, verified_timecard)
       expect(active_timecards).not_to include(cancelled_timecard)
     end
   end
 
-  describe "creation" do
-    it "has a valid factory" do
-      timecard = build(:timecard, person: @cj)
-      expect(timecard).to be_valid
+  describe 'older_than scope' do
+    it 'returns a chainable relation' do
+      expect(described_class.older_than(1)).to be_a_kind_of(ActiveRecord::Relation)
     end
+    
+    context 'given 2 timecards with different start_times' do
+      before(:example) do
+        @timecard_with_7hours = create(:timecard, person: @cj,
+          start_time: 7.hours.ago, end_time: 1.hour.ago)
+        @timecard_with_2hours = create(:timecard, person: @cj,
+          start_time: 2.hours.ago, end_time: 1.hour.ago)
+      end
 
-    it "requires a person" do
-      timecard = build(:timecard, person: nil)
-      expect(timecard).not_to be_valid
+      it 'returns timecards with a start_time older than 3 hours' do
+        expect(described_class.older_than(3)).to contain_exactly(
+          @timecard_with_7hours)
+      end
+
+      it 'returns timecards with a start_time older than 1 hour' do
+        expect(described_class.older_than(1)).to contain_exactly(
+          @timecard_with_7hours, @timecard_with_2hours)
```

Comment:
This makes the duration more important, as the shorter card is actually one hour. This could lead to an intermittent test, as it's right at the line. Maybe change the end times to `30.minutes.ago` ?

Reply:
Heh, I mean to describe the "age" of the start_time attribute, I'll change these to x_hours_old_timecard

Reply:
Right, but I would like to have a wider margin for the assertion too.